apps/umpire/README.md

```diff
@@ -0,0 +1,94 @@
+2. Log these and other events with a timestamp and a duration since the last event.
+3. Allow the data to be viewed on the watch and on a BLE-paired mobile device as structured data.
+
+## Modules ##
+
+The app is split into three screens shown to the umpire:
+1. Main Menu
+2. In-play Screen
+3. Log viewer
+
+There is an additional module to allow settings to be changed via the watch Settings menu; the number of balls in the over and the number of overs in the innings are configurable, so that 15 eight-ball overs, T20, 40-50 overs and The Hundred can all be supported.
+
+A further module of the app is the web page used to view the log file in the app loader. The file can be deleted or a copy downloaded.
+
+## Interaction ##
+
+Bangle.js 2 app interactions are governed by the Bangle and Espruino (E) libraries. This app uses the following patterns:
+
+### Swipe and Button ###
+
+The in-play screen detects swipe and button events using Bangle.setUI. This appears reliable. 
+
+When in-play, use of the button increments the (fairly delivered) ball count. The app does not suppress the screen lock so by default it requires the umpire to press once to unlock the watch and a second press to log the ball. Bangle.buzz is used to give positive feedback when the app takes action so that the screen does not need to be looked at.
+
+After logging the third-to-last ball in the over, the app buzzes twice. This is typically when the umpire needs to check the balls remaining with the other umpire, in case they have not logged every fair delivery.
+
+After the second-to-last ball the app gives one long buzz to remind the umpire that the next ball will close the over (if fairly delivered).
+
+Logging the final ball triggers the change of over. 
+
+When the last over of the innings is complete the screen shows "END" but allows play to continue to be logged until the new innings is triggered.
+
+Whilst the in-play screen is displayed swiping will cause the following actions to occur:
+- **Swipe Up** performs the same action as the Button press, incrementing the ball count.
+- **Swipe Down** decrements the ball count and logs the action as a "Correction".
+- **Swipe Right** shows the log viewer. 
+- **Swipe Left** shows the main menu.
+
+### Confirmation Prompts ###
+
+When choosing "Wicket" or "Recall" from the main menu, E.showPrompt is used to ask for positive confirmation before incrementing or decrementing the wickets, respectively. Those functions also increment or decrement the ball count. E.showPrompt offers relatively small tap targets on the Bangle.js 2 screen but appears stable.
+
+### Scrollers ###
+
+The main menu and log viewer both use E.showScroller to display a scrollable list of tappable items. This is in preference to E.showMenu which, at the time of writing, has bugs affecting where the touch event is detected in the list of menu items (one lower than it should).
```

Comment:
Probably worth opening an issue for this, I remember seeing it when widgets are brought in - @gfwilliams @thyttan have either of you seen this before?

Reply:
I'm not sure. A week or two ago I maybe noticed something similar with showScroller. I had initiated it with widgets showing but immediately after hid them, and I think then drawing was a bit wonky and maybe input as well. Not sure.

Reply:
Interesting - well I'll not block this PR because of it but will keep my eye out for it happening

Reply:
@AndySmee if you could provide more info on this in an issue (when using 2v26 firmware, and making sure your touchscreen is calibrated) that'd be hugely helpful. E.showMenu uses E.showScroller under the hood anyway, so it's kind of surprising there's an issue.

But we use E.showMenu for so much on Bangle.js (Settings/etc) that if there's an issue I'd really like to get it sorted rather than having app developers trying not to use it!